### DISCUSSION TITLE
Simulator does not open

### DISCUSSION THREAD

**Discussion by hunkymcstudleypants** [inkstitch/inkstitch#1527] [Q&A]
I installed inkstitch recently and the Simulator / Realistic Preview has never worked. I click **Extensions** -> **Ink/Stitch** -> **Visualize and Export** -> **Simulator / Realistic Preview**, then my mouse pointer changes to a spinning wheel for about 1 second, then back to a pointer, but nothing else happens. There are no errors or dialog, just nothing. I have tried this with objects/paths I've created in Inkscape, with SVG's imported from GIMP, and followed along with the Inkstitch beginner tutorial playlist on YouTube (for which the Inkscape UI in the screen recording doesn't match what I see in Inkscape)  - always the same outcome: nothing. Am I missing something?

Capturing log messages shows only this:
`StyleDialog::_handleSelectionChanged()
StyleDialog::readStyleElement
StyleDialog::_getStyleTextNoded
Found resource file 'dialog-css.glade' in system directory:
	/Applications/Inkscape.app/Contents/Resources/share/inkscape/ui/dialog-css.glade
StyleDialog::_handleSelectionChanged()
StyleDialog::readStyleElement
StyleDialog::_getStyleTextNoded
Found resource file 'dialog-css.glade' in system directory:
	/Applications/Inkscape.app/Contents/Resources/share/inkscape/ui/dialog-css.glade
`

**Comment by tyrosinase** [comment]
What operating system are you using, and how did you install it?

**Comment by hunkymcstudleypants** [comment]
i'm on macOS Monterey 12.0.1. i downloaded the zip from inkstitch.org and extracted it in my extensions directory. 

when i set the params, the preview / simulator displays with no issue. it's only when i specifically select **Simulator / Realistic Preview** that nothing happens.

**Reply by jfk1972** [reply]
@hunkymcstudleypants I get the same behavior with V2.0 running on Windows 10 32-bit. Windows does not have the spinning wheel, so I simply get nothing when I invoke the simulator.

**Reply by hunkymcstudleypants** [reply]
Does anyone have any ideas for @jfk1972 and me? 

This issue is only slightly annoying. Not nearly as frustrating as the inconsistent-yet-persistent "border crosses itself" error that comes up when the border certainly doesn't cross itself.

**Comment by tyrosinase** [comment]
In the previous version the border-crossing error sometimes came up when there were very sharp points in a shape; as I understand it (without having looked at the code) the error gets triggered when both ends of a stitch end up outside the bounds. Rotating the stitching sometimes helps, or just blunting the points (since they're not going to get sewn anyway).

(I'm on Linux so I don't have any more input on the other problems, other than it sounds like what happens when there's an error... except without actually displaying the error, which I don't think I've had happen.)

**Reply by hunkymcstudleypants** [reply]
The border crossing error comes up on perfect circle ellipses drawn in inkscape. There are no sharp points in that scenario. I've just come to accept that inkscape is riddled with bugs that will likely never be addressed due to inkscape's lack of resources. I was using it to generate g-code for CNC milling as well, but it is far too frustrating a software to work with. 
It's FOSS, so I can't expect it to be perfect and am grateful for the devs who volunteer their time to projects like inkscape, but it's unusable for digitizing embroidery or generating g-code.
